strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/NpmArtifactCoordinates.java

```diff
@@ -0,0 +1,152 @@
+package org.carlspring.strongbox.artifact.coordinates;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import org.semver.Version;
+import org.springframework.util.Assert;
+
+/**
+ * This class is an {@link ArtifactCoordinates} implementation for npm
+ * artifacts.
+ * <br>
+ * See <a href="https://docs.npmjs.com/files/package.json">Official npm
+ * package specification</a>.
+ * 
+ * @author sbespalov
+ *
+ */
+@SuppressWarnings("serial")
+@XmlRootElement(name = "npmArtifactCoordinates")
+@XmlAccessorType(XmlAccessType.NONE)
+public class NpmArtifactCoordinates extends AbstractArtifactCoordinates
+{
+    public static final String NPM_NAME_REGEX = "[a-z0-9][\\w-.]*";
+    public static final String NPM_EXTENSION_REGEX = "\\.tgz";
+    public static final String NPM_PACKAGE_PATH_REGEX = "(@?" + NPM_NAME_REGEX + ")/(" + NPM_NAME_REGEX + ")/"
+            + NPM_NAME_REGEX + "-(.+?(?=" + NPM_EXTENSION_REGEX + "))" + NPM_EXTENSION_REGEX;
+
+    private static final Pattern NPM_NAME_PATTERN = Pattern.compile(NPM_NAME_REGEX);
+    private static final Pattern NPM_PATH_PATTERN = Pattern.compile(NPM_PACKAGE_PATH_REGEX);
+
+    private static final String SCOPE = "scope";
+    private static final String NAME = "name";
+    private static final String VERSION = "version";
+
+    public NpmArtifactCoordinates()
+    {
+        defineCoordinates(SCOPE, NAME, VERSION);
+    }
+
+    public NpmArtifactCoordinates(String scope,
+                                  String name,
+                                  String version)
+    {
+        setScope(scope);
+        setName(name);
+        setVersion(version);
+    }
+
+    public String getScope()
+    {
+        return getCoordinate(SCOPE);
+    }
+
+    public void setScope(String scope)
+    {
+        if (scope == null)
+        {
+            return;
+        }
+        Assert.isTrue(scope.startsWith("@"), "Scope should starts with '@'.");
+        setCoordinate(SCOPE, scope);
+    }
+
+    public String getName()
+    {
+        return getCoordinate(NAME);
+    }
+
+    public void setName(String name)
+    {
+        Matcher matcher = NPM_NAME_PATTERN.matcher(name);
+        Assert.isTrue(matcher.matches(),
+                      "NPM artifact name shoud fit specification (https://docs.npmjs.com/files/package.json#name).");
```

Comment:
"The artifact's name should follow the NPM specification (https://docs.npmjs.com/files/package.json#name)."

Reply:
ok